src/main/scala/org/broadinstitute/dsde/firecloud/service/LibraryService.scala

```diff
@@ -95,4 +97,20 @@ class LibraryService (protected val userInfo: UserInfo, val rawlsDAO: RawlsDAO)(
       }
     }
   }
+
+  def setWorkspaceIsPublished(ns: String, name: String, value: Boolean): Future[PerRequestMessage] = {
+    rawlsDAO.getWorkspace(ns, name) flatMap { workspaceResponse =>
+      // verify owner on workspace
+      if (!workspaceResponse.accessLevel.contains("OWNER")) {
+        Future(RequestCompleteWithErrorReport(Forbidden, "must be an owner"))
+      } else {
+        val operations: Seq[AttributeUpdateOperation] = Seq()
+        // either add or remove the library attribute
+        if (value) operations :+ AddUpdateAttribute("library", AttributeString("published"))
+        else operations :+ RemoveAttribute("library")
```

Comment:
you're creating an empty Seq, then creating a new copy of that Seq with a value appended. It would be more efficient to create it once. Does this work?
```
val operations: Seq[AttributeUpdateOperation] = if (value)
  Seq(AddUpdateAttribute("library", AttributeString("published")))
else
  Seq(RemoveAttribute("library"))
```

also, add a TODO comment here - once we support boolean attributes, it would be better for the value to be true/false instead of the string "published". Unless you think we need different states for publishing, e.g. "pending, published, approved, etc"?